src/lazyexpression.jl

```diff
@@ -24,32 +24,39 @@ end
     end
 end
 
-
-# expression macro
-macro expression(expr)
-    # We call expand(expr) to lower the code and make some useful conversions like:
-    #   * [x; y] turns from Expr(:vcat, :x, :y) into Expr(:call, :vcat, :x, :y)
-    #   * x .+ y turns from Expr(:call, :.+, :x, :y) into Expr(:call, :broadcast, :+, :x, :y) (on v0.6)
-    postwalk(expand(expr)) do x
-        if @capture(x, f_(args__))
-            :(wrap(SimpleQP.optimize_toplevel(SimpleQP.LazyExpression($f, $(args...)))))
+lazy_wrap(x) = esc(x)
+
+function lazy_wrap(x::Expr)
+    if x.head == :line
+        return esc(x)
+    elseif x.head == :$ && length(x.args) == 1
+        return esc(x.args[1])
+    elseif x.head == :block
+        return esc(Expr(x.head, lazy_wrap.(x.args)...))
```

Comment:
Ah, I see now that I was mistaken in https://github.com/tkoolen/SimpleQP.jl/issues/30#issuecomment-400261322. I had missed that `lazy_wrap(x) = esc(x)`, so I think the `esc` around the `Expr` here is not necessary (and actually bad).

Reply:
Oh, you're right, that's too much escaping.